internal/docs/UpgradeotelCol.md

```diff
@@ -0,0 +1,73 @@
+prometheus-receiver
+* copy over new folder
+* go.mod rename module
+go.mod remove replacements at the end
+Find new version of github.com/prometheus/prometheus. Put this version in the file /otelcollector/opentelemetry-collector-builder/PROMETHEUS_VERSION
+* delete testdata directory
+* metrics_receiver.go: rename internal package "github.com/gracewehner/prometheusreceiver/internal"
+* metrics_receiver.go: add webhandler code in initPrometheusComponents() or Start() function
+* metrics_receiver.go: add extra import packages at the top
+* metrics_receiver.go: add constants at the top
+internal/otlp_transaction.go: in Append() function before if len(t.externalLabels) != 0 (currently line 92) add labels = labels.Copy()
+prom-config-validator-builder
+* update go.mod to new collector version for all components
+* try to build to check for any breaking changes to the interfaces used: run make
+opentelemetry-collector-builder
+go mod tidy
+module github.com/gracewehner/prometheusreceiver
+    "github.com/prometheus/client_golang/prometheus"
+    "github.com/prometheus/common/version"
+    "github.com/prometheus/prometheus/web"
+    // Use same settings as Prometheus web server
+    maxConnections = 512
+    readTimeoutMinutes = 10
+    // Setup settings and logger and create Prometheus web handler
+    webOptions := web.Options{
+        ScrapeManager: r.scrapeManager,
+        Context: ctx,
+        ListenAddress: ":9090",
+        ExternalURL: &url.URL{
+            Scheme: "http",
+            Host:   "localhost:9090",
+            Path:   "",
+        },
+        RoutePrefix:    "/",
+        ReadTimeout: time.Minute * readTimeoutMinutes,
+        PageTitle: "Prometheus Receiver",
+        Version: &web.PrometheusVersion{
+            Version:   version.Version,
+            Revision:  version.Revision,
+            Branch:    version.Branch,
+            BuildUser: version.BuildUser,
+            BuildDate: version.BuildDate,
+            GoVersion: version.GoVersion,
+        },
+        Flags: make(map[string]string),
+        MaxConnections: maxConnections,
+        IsAgent: true,
+        Gatherer:   prometheus.DefaultGatherer,
+    }
+    go_kit_logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+    webHandler := web.New(go_kit_logger, &webOptions)
+    listener, err := webHandler.Listener()
+    if err != nil {
+        return err
+    }
+    // Pass config and let the web handler know the config is ready.
+    // These are needed because Prometheus allows reloading the config without restarting.
+    webHandler.ApplyConfig(r.cfg.PrometheusConfig)
+    webHandler.SetReady(true)
+    // Uses the same context as the discovery and scrape managers for shutting down
+    go func() {
+        if err := webHandler.Run(ctx, listener, ""); err != nil {
+            r.settings.Logger.Error("Web handler failed", zap.Error(err))
+            host.ReportFatalError(err)
+        }
+    }()
```

Comment:
Thanks Soham, could you add the ` ``` ` markdown code formatting for this?

Also it would be great to attach a link to the video of our teams call too just as another future reference

Reply:
Thanks Grace! I will update.

Reply:
I have added shareable link to the screenshare, people within MS can access this link.